pkg/rpc/client/client.go

```diff
@@ -40,6 +40,10 @@ type Client struct {
 	// cache is mostly filled in during Init(), but can also be updated
 	// during regular Client lifecycle.
 	cache cache
+
+	reqIDsLock  sync.RWMutex
+	reqIDs      map[int]struct{}
+	latestReqID int
```

Comment:
`uint64`?

Reply:
And if we're to use an atomic value we might as well avoid another `RWMutex` to protect it.

Reply:
But what about https://github.com/nspcc-dev/neo-go/pull/2367#discussion_r811809151?

Reply:
Well, that is valid as long as have a `reqIDs` map, but if you don't have it, you might as well avoid using a mutex.